android/src/main/java/com/mrousavy/camera/utils/ImageProxy.save.kt

```diff
@@ -40,6 +40,39 @@ fun flip(imageBytes: ByteArray, imageWidth: Int): ByteArray {
   return holder + subArray
 }
 
+fun fixOrientation(imageBytes: ByteArray, orientation: Int, flipHorizontally: Boolean): ByteArray {
+  val bitmap = BitmapFactory.decodeByteArray(imageBytes, 0, imageBytes.size)
+  val matrix = Matrix()
+  val oneEight = 45 // (360/8)
+  val exif = ExifInterface(imageBytes.inputStream())
+
+  when (orientation) {
+    //  landscape left
+    in oneEight..3*oneEight -> {
+      matrix.setRotate(180f)
+      exif.setAttribute(ExifInterface.TAG_ORIENTATION, ExifInterface.ORIENTATION_ROTATE_180.toString())
+    }
+    //  upside down
+    in 3*oneEight..5*oneEight-> {
+      matrix.setRotate(if (flipHorizontally) -270f else 270f)
+      exif.setAttribute(ExifInterface.TAG_ORIENTATION, ExifInterface.ORIENTATION_ROTATE_270.toString())
+    }
+    // landscape right
+    in 5*oneEight..7*oneEight -> {
```

Comment:
can you explain how these ranges were calculated?

Reply:
the orientation sensor returns a number between 0-359,
from what iv'e seen in old lib, it looks like changes from portrait to landscape (for example) in the middle, like when phone is pointing half  up half to the side.
so it's jut like 1/8 of 360 to the right. so each position has 90 degrees of existence but its not from zero, its:
portrait: 315-45
landscape: 45-135
etc...